crates/rust-analyzer/src/config.rs

```diff
@@ -234,13 +235,14 @@ impl Config {
                 }
                 // otherwise configure command customizations
                 _ => {
-                    if let Some(FlycheckConfig::CargoCommand { command, extra_args, all_targets, all_features })
+                    if let Some(FlycheckConfig::CargoCommand { command, extra_args, all_targets, all_features, features })
                         = &mut self.check
                     {
                         set(value, "/checkOnSave/extraArgs", extra_args);
                         set(value, "/checkOnSave/command", command);
                         set(value, "/checkOnSave/allTargets", all_targets);
                         set(value, "/checkOnSave/allFeatures", all_features);
```

Comment:
Let's either:

* copy `allFeatures` from `cargo` and remove it from `package.json`
* add `checkOnSave.features` to `package.json` and use some kind of priority shceme for both (ie, if they are not set explicitelly in user config, we use values from cargo section).